packages/react-dom/src/client/ReactInputSelection.js

```diff
@@ -116,7 +116,13 @@ export function restoreSelection(priorSelectionInformation) {
   const curFocusedElem = getActiveElementDeep();
   const priorFocusedElem = priorSelectionInformation.focusedElem;
   const priorSelectionRange = priorSelectionInformation.selectionRange;
-  if (curFocusedElem !== priorFocusedElem && isInDocument(priorFocusedElem)) {
+  // With disabled elements, they are not focusable
+  if (
+    priorFocusedElem &&
+    !priorFocusedElem.disabled &&
```

Comment:
Are there ever cases where reading this would throw?

Reply:
Hi @gaearon, I'm new to react contribution so I'm kinda unsure if it may behave strangely somewhere. However, I'm confident to create a PR just because I believe this change is minimal and safe enough to not throw any errors and the passed tests should also be trustworthy.

Reply:
Since the master branch has been renamed, there are conflicts in this PR so I've added a new PR for this at #22256, please take a look :)